channel/consensus_channel/consensus_channel.go

```diff
@@ -224,11 +267,21 @@ func (c *ConsensusChannel) Propose(add Add, sk []byte) (SignedProposal, error) {
 // sign constructs a state.State from the given vars, using the ConsensusChannel's constant
 // values. It signs the resulting state using pk.
 func (c *ConsensusChannel) sign(vars Vars, pk []byte) (state.Signature, error) {
+	signer := crypto.GetAddressFromSecretKeyBytes(pk)
```

Comment:
If we like this check, might be nice to spread it around -- i.e. into the `Channel` class.

Reply:
An improvement would be to pass around a `*Signer` instead that stores a `pk` and an `address`.

`Signer` could implement `Address()`, which computes its address on the first call and remembers it for later. This would allow the kind of check we're doing here, but only incur the cost once.

Or, `Signer` could implement `Sign(me types.Address)` and return an error if the signer's private key doesn't match `me`. Again, it could compute the correct address once and store it.